### PR TITLE
Fix the WPT exporter

### LINKED ISSUE #30869: WPT export is broken
https://github.com/servo/servo/actions/runs/7180806387/job/19553860946

```
INFO:root:  → Execution (cwd='./wpt'): git checkout -b servo_export_30767
INFO:root:  → Execution (cwd='./wpt'): git fetch --unshallow origin
INFO:root:  → Execution (cwd='./wpt'): git push -f ***github.com/servo-wpt-sync/web-platform-tests.git servo_export_30767
INFO:root:  → Execution (cwd='./wpt'): git checkout main
WARNING:root:Process execution failed with output:
error: pathspec 'main' did not match any file(s) known to git

INFO:root:  → Request: POST repos/web-platform-tests/wpt/pulls
INFO:root:  → Request JSON: {'title': 'Add initial support for offscreen rendering', 'head': 'servo-wpt-sync:servo_export_30767', 'base': 'main', 'body': 'This patch makes it possible to direct the output of the compositor to an offscreen OpenGL framebuffer object, rather than the OpenGL backbuffer or an image in IpcSharedMemory or a png file.\r\n\r\nWe also make servoshell use this iff the minibrowser is enabled, and rework CompositeTarget to allow the embedder to specify the target and output file if any when initialising Servo.\r\n\r\nCompositeTarget before:\r\n\r\n- Window draws to the OpenGL backbuffer only\r\n- WindowAndPng draws to an uncompressed image in shared memory only\r\n    - the name was misleading; it’s neither WindowAnd nor Png\r\n- PngFile draws to IOCompositor.output_file only (which must be Some) and exits\r\n\r\nCompositeTarget after:\r\n\r\n- Window draws to the OpenGL backbuffer only\r\n- Fbo draws to an offscreen OpenGL framebuffer object (IOCompositor::output_framebuffer_id)\r\n- SharedMemory draws to an uncompressed image in shared memory only\r\n- PngFile(String) draws to a PNG file at the given path only and exits\r\n\r\nManually verified working on:\r\n- [x] Windows\r\n- [x] macOS\r\n- [x] Linux (X11)\r\n- [x] Linux (Wayland) — no worse than before\r\n    - #<!-- nolink -->30826\r\n    - #<!-- nolink -->30827\r\n\r\nReviewed in servo/servo#30767', 'maintainer_can_modify': False}
[...]
ERROR:root:Got unexpected 422 response: {"message":"Validation Failed","errors":[{"resource":"PullRequest","field":"base","code":"invalid"}],"documentation_url":"https://docs.github.com/rest/pulls/pulls#create-a-pull-request"}
Traceback (most recent call last):
  File "/home/runner/work/servo/servo/servo/python/wpt/exporter/__init__.py", line 195, in run
    run.run()
  File "/home/runner/work/servo/servo/servo/python/wpt/exporter/__init__.py", line 106, in run
    step.run(self)
  File "/home/runner/work/servo/servo/servo/python/wpt/exporter/step.py", line 292, in run
    pull_request = self.target_repo.open_pull_request(
  File "/home/runner/work/servo/servo/servo/python/wpt/exporter/github.py", line 112, in open_pull_request
    response = authenticated(self.sync, "POST", self.pulls_url, json=data)
  File "/home/runner/work/servo/servo/servo/python/wpt/exporter/github.py", line 47, in authenticated
    raise ValueError(
ValueError: Got unexpected 422 response: {"message":"Validation Failed","errors":[{"resource":"PullRequest","field":"base","code":"invalid"}],"documentation_url":"https://docs.github.com/rest/pulls/pulls#create-a-pull-request"}
Error: Process completed with exit code 1.
```

### PR DESCRIPTION
**Pull request by delan**
#30788 updates branch references from “master” to “main”, but [web-platform-tests/wpt](https://github.com/web-platform-tests/wpt) and [servo-wpt-sync/web-platform-tests](https://github.com/servo-wpt-sync/web-platform-tests) still use “master”, so the WPT exporter now fails to check out and open pull requests against “main”.

This patch makes the default branch name in GithubRepository configurable, set to main in Servo and master in WPT. We also fix the tests so they can now run on NixOS.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #30869 (GitHub issue number if applicable)

### CODE REVIEW

**Comment by sagudev** [comment]
WPT export tests are failing (run with `./mach test-scripts`)

**Comment by mrobinson** [comment]
Here's a fix for the tests that makes the main branch name configurable:

```diff
diff --git a/python/wpt/exporter/__init__.py b/python/wpt/exporter/__init__.py
index c9b5e5d028..8d942f0fee 100644
--- a/python/wpt/exporter/__init__.py
+++ b/python/wpt/exporter/__init__.py
@@ -142,9 +142,9 @@ class WPTSync:
     suppress_force_push: bool = False
 
     def __post_init__(self):
-        self.servo = GithubRepository(self, self.servo_repo)
-        self.wpt = GithubRepository(self, self.wpt_repo)
-        self.downstream_wpt = GithubRepository(self, self.downstream_wpt_repo)
+        self.servo = GithubRepository(self, self.servo_repo, "main")
+        self.wpt = GithubRepository(self, self.wpt_repo, "master")
+        self.downstream_wpt = GithubRepository(self, self.downstream_wpt_repo, "master")
         self.local_servo_repo = LocalGitRepo(self.servo_path, self)
         self.local_wpt_repo = LocalGitRepo(self.wpt_path, self)
 
diff --git a/python/wpt/exporter/github.py b/python/wpt/exporter/github.py
index 31fa5bb93c..5e46f4a148 100644
--- a/python/wpt/exporter/github.py
+++ b/python/wpt/exporter/github.py
@@ -55,9 +55,10 @@ class GithubRepository:
     This class allows interacting with a single GitHub repository.
     """
 
-    def __init__(self, sync: WPTSync, repo: str):
+    def __init__(self, sync: WPTSync, repo: str, main_branch_name: str):
         self.sync = sync
         self.repo = repo
+        self.main_branch_name = main_branch_name
         self.org = repo.split("/")[0]
         self.pulls_url = f"repos/{self.repo}/pulls"
 
@@ -74,7 +75,7 @@ class GithubRepository:
         self, branch: GithubBranch
     ) -> Optional[PullRequest]:
         """If this repository has an open pull request with the
-        given source head reference targeting the master branch,
+        given source head reference targeting the main branch,
         return the first matching pull request, otherwise return None."""
 
         params = "+".join([
@@ -105,7 +106,7 @@ class GithubRepository:
         data = {
             "title": title,
             "head": branch.get_pr_head_reference_for_repo(self),
-            "base": "master",
+            "base": self.main_branch_name,
             "body": body,
             "maintainer_can_modify": False,
         }
diff --git a/python/wpt/test.py b/python/wpt/test.py
index f30e01eb79..452a5587ca 100644
--- a/python/wpt/test.py
+++ b/python/wpt/test.py
@@ -644,17 +644,17 @@ def setUpModule():
         suppress_force_push=True,
     )
 
-    def setup_mock_repo(repo_name, local_repo):
+    def setup_mock_repo(repo_name, local_repo, main_branch_name: str):
         subprocess.check_output(
             ["cp", "-R", "-p", os.path.join(TESTS_DIR, repo_name), local_repo.path])
-        local_repo.run("init", "-b", "main")
+        local_repo.run("init", "-b", main_branch_name)
         local_repo.run("add", ".")
         local_repo.run("commit", "-a", "-m", "Initial commit")
 
     logging.info("=" * 80)
     logging.info("Setting up mock repositories")
-    setup_mock_repo("servo-mock", SYNC.local_servo_repo)
-    setup_mock_repo("wpt-mock", SYNC.local_wpt_repo)
+    setup_mock_repo("servo-mock", SYNC.local_servo_repo, "main")
+    setup_mock_repo("wpt-mock", SYNC.local_wpt_repo, "master")
     logging.info("=" * 80)
  ```

**Comment by delan** [comment]
Thanks! 212e6384a74ae4a8807a587007fe938e4ea1c7c5 applies @mrobinson’s patch, and 64c937c03f9411e0c3364de9292395334bf961ab fixes the tests so they run on NixOS.

File: python/wpt/test.py [hunk 1]
```diff
@@ -644,17 +644,17 @@ def setUpModule():
         suppress_force_push=True,
     )
 
-    def setup_mock_repo(repo_name, local_repo):
+    def setup_mock_repo(repo_name, local_repo, main_branch_name: str):
```
Comment (sagudev) [inline python/wpt/test.py:647]: `default_branch` is used in JSON of expectations.

**Review by sagudev — APPROVED** [review]

File: python/wpt/test.py (hunk 1 above)
Comment (delan) [reply python/wpt/test.py]: good point, renamed in d0808c849b2c0a2f85551689de43ad06964bec5e

**Pull request merged**